feat(trace-view): For !javascript add a common connect services (#3402)
getsentry/sentry-docs: src/platforms/common/performance/connect-services.mdx

redirect_from:
---

> To connect backend and frontend transactions into a single coherent trace, Sentry uses a `trace_id` value that is propagated between frontend and backend services. Depending on the circumstance, this ID is transmitted either in a request header or in an HTML `<meta>` tag. Linking transactions in this way makes it possible to navigate between them in sentry.io to better understand how the different parts of your system are affecting one another. You can learn more about this model in our [Distributed Tracing](/product/performance/distributed-tracing/) docs.

Reviewer: Should sentry.io be a link?

> To connect backend and frontend transactions into a single coherent trace, Sentry uses a `trace_id` value which is propagated between frontend and backend services. Depending on the circumstance, this id is transmitted either in a request header or in an HTML `<meta>` tag. Linking transactions in this way makes it possible for you to navigate between them in the Sentry UI, so you can better understand how the different parts of your system are affecting each other. You can learn more about this model in our [Distributed Tracing](/product/performance/distributed-tracing/) docs.

Reviewer 2: ```suggestion
To connect backend and frontend transactions into a single coherent trace, Sentry uses a `trace_id` value that is propagated between frontend and backend services. Depending on the circumstance, this ID is transmitted either in a request header or in an HTML `<meta>` tag. Linking transactions in this way makes it possible to navigate between them in sentry.io to better understand how the different parts of your system are affecting one another. You can learn more about this model in our [Distributed Tracing](/product/performance/distributed-tracing/) docs.
```

### Navigation and Other XHR Requests

Once a page is loaded, any requests it makes (and any requests your backend makes as a result) are linked through a request header.

> All of Sentry's tracing-related integrations (`BrowserTracing`, `Http`, and `Express`) either generate or pick up and propagate the trace header automatically as appropriate, for all transactions and spans which they generate.

Reviewer 2: ```suggestion
All of Sentry's tracing-related integrations (`BrowserTracing`, `Http`, and `Express`) either generate or pick up and propagate the trace header automatically as appropriate for all transactions and spans that they generate.
```

The JavaScript SDK will only attach the trace header to outgoing HTTP requests whose destination contains a string in or matches a regex in the <PlatformLink to="/performance/included-instrumentation/#tracingorigins">tracingOrigin</PlatformLink> list.

> You can also attach and read the header yourself whenever you've manually created either a transaction or a span and it makes sense to do so. The header's name is `sentry-trace` and its value is obtained by calling `span.toTraceparent()` (or the equivalent), where `span` is either the relevant transaction or any of its children.

Reviewer: This paragraph is confusing. I feel after moving it from the other section we are now mixing up concepts. The call to `span.toTraceparent()` is not useful when you *create* manual spans. It returns the value for the trace propagation header for outgoing requests originating from the given span.

Author: Glad to get your review on this, so I'm honestly not too familiar with this method, and using this to manually set headers was what I thought the original text here was trying to call out.

So for an end user, would they have any other reason to know about or call `span.toTraceparent` other than trying to manually set the header themselves?

> You can also attach and read the header yourself, in any case in which you've manually created either a transaction or a span and it makes sense to do so. The header's name is `sentry-trace` and its value is obtained by calling `span.toTraceparent()` (or the equivalent), where `span` is either the relevant transaction or any of its children.

Reviewer 2: ```suggestion
You can also attach and read the header yourself whenever you've manually created either a transaction or a span and it makes sense to do so. The header's name is `sentry-trace` and its value is obtained by calling `span.toTraceparent()` (or the equivalent), where `span` is either the relevant transaction or any of its children.
```
